core/src/wallet/ethereum/EthereumLikeAccount.cpp

```diff
@@ -317,9 +342,41 @@ namespace ledger {
                         return keychain->contains(addr);
                     };
 
-                    //Get operations related to an account
+                    // Get operations related to an account
                     OperationDatabaseHelper::queryOperations(sql, uid, operations, filter);
 
+                    // Get internal transactions for every operations related to an account and
+                    // turn them into logical operation to affect the balance
+                    {
+                      // a buffer for allocating internal operations
+                      std::vector<InternalTransaction> internalTransactions;
+                      std::vector<Operation> internalOperations;
+
+                      // this loop get all the internal transactions related to a given ETH
+                      // operation and will transform them into regular Operation
+                      for (auto const& operation : operations) {
+                        // get this operation’s internal tx
+                        internalTransactions.clear();
+                        getInternalTransactions(sql, operation, internalTransactions);
```

Comment:
A little bit afraid about performance of this. For each operation we will do request to DB.
Maybe we need to redefine what is inside `Operation` so getOperation will return everything with one select with JOIN

Reply:
It’s a cold path so I’d say it’s okay.

Reply:
But I agree that we could do it at once. I could write such a SQL query actuall and bake it in `getOperationsWithInternal`.